gson/src/test/java/com/google/gson/internal/bind/util/ISO8601UtilsTestCase.java

```diff
@@ -0,0 +1,205 @@
+   * This test covers `int indexOfNonDigit(String)' block 6 (line 346)
+   * This test covers `int indexOfNonDigit(String)' block 7 (line 346)
+   * This test covers `int indexOfNonDigit(String)' block 8 (line 347)
+   * This test covers `int indexOfNonDigit(String)' block 10 (line 347)
+   * This test covers `int indexOfNonDigit(String)' block 11 (line 345)
+   * This test covers `int indexOfNonDigit(String)' block 12 (line 347)
+   * This test covers `int indexOfNonDigit(String)' block 13 (line 345)
+   *
+   */
+
+  @Test
+  public void indexOfNonDigitInputNotNullZeroOutputPositive()
+      throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {
+
+    // Arrange
+    final String string = "2`88888888";
+    final int offset = 0;
+
+    // Act
+    final Class<?> classUnderTest =
+        Reflector.forName("com.google.gson.internal.bind.util.ISO8601Utils");
+    final Method methodUnderTest = classUnderTest.getDeclaredMethod(
+        "indexOfNonDigit", Reflector.forName("java.lang.String"), Reflector.forName("int"));
+    methodUnderTest.setAccessible(true);
+    final int retval = (Integer)methodUnderTest.invoke(null, string, offset);
+
+    // Assert result
+    Assert.assertEquals(1, retval);
+  }
+
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *
+   */
+
+  @Test
+  public void indexOfNonDigitInputNotNullZeroOutputZero()
+      throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {
+
+    // Arrange
+    final String string = "";
+    final int offset = 0;
+
+    // Act
+    final Class<?> classUnderTest =
+        Reflector.forName("com.google.gson.internal.bind.util.ISO8601Utils");
+    final Method methodUnderTest = classUnderTest.getDeclaredMethod(
+        "indexOfNonDigit", Reflector.forName("java.lang.String"), Reflector.forName("int"));
+    methodUnderTest.setAccessible(true);
+    final int retval = (Integer)methodUnderTest.invoke(null, string, offset);
+
+    // Assert result
+    Assert.assertEquals(0, retval);
+  }
+
+  /*
+   * Test generated by Diffblue Deeptest.This test covers `int
+   * indexOfNonDigit(String)' block 2 (line 345)
+   * This test covers `int indexOfNonDigit(String)' block 3 (line 345)
+   * This test covers `int indexOfNonDigit(String)' block 4 (line 345)
+   * This test covers `int indexOfNonDigit(String)' block 6 (line 346)
+   * This test covers `int indexOfNonDigit(String)' block 7 (line 346)
+   * This test covers `int indexOfNonDigit(String)' block 8 (line 347)
+   * This test covers `int indexOfNonDigit(String)' block 9 (line 347)
+   * This test covers `int indexOfNonDigit(String)' block 12 (line 347)
+   *
+   */
+
+  @Test
+  public void indexOfNonDigitInputNotNullZeroOutputZero2()
+      throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {
+
+    // Arrange
+    final String string = "\"`((((((((";
+    final int offset = 0;
```

Comment:
Is there no test for non-zero or negative offset?

Reply:
There are no tests of this method with a non-zero offset.